### PR TITLE
Added NewDefaultCommonFlagsWithoutUsage.  Simplified

### PR DESCRIPTION
**Pull request by michaelestrin**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://wiki.edgexfoundry.org/display/FA/Contributor%27s+Guide
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [x] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [x] Other... Please describe:

Added new factory function to be called when no additional usage is needed.  Interface segregation principle.  Code clean up.

## What is the current behavior?

Issue Number: https://github.com/edgexfoundry/go-mod-bootstrap/issues/44

## Does this PR introduce a breaking change?

- [x] Yes
- [ ] No

## Other information

### CODE REVIEW

File: bootstrap/commandline/commandline.go [hunk 1]
```diff
@@ -25,8 +25,8 @@ const (
 	DefaultConfigFile     = "configuration.toml"
 )
 
-// CommonFlags is an interface that defines AP for the common command-line flags used by most EdgeX services
-type CommonFlags interface {
+// Flags is an interface that defines AP for the common command-line flags used by most EdgeX services
```
Comment (lenny-intel) [inline bootstrap/commandline/commandline.go]: I don't agree with this renaming.  `CommonFlags` is explicit that encapsulates the common flags and `Flags` doesn't . From a usage POV, `Flags` just looks to be an alternative implementation and doesn't imply that it actually add all the common flags.

Comment (lenny-intel) [reply bootstrap/commandline/commandline.go]: Also, per @tsconn23, bootstrap/commandline/commandline.go is name stuttering. I googled this and didn't find anything on filename stuttering. I did find package/struct stuttering guidelines, which isn't an issue here. ;-)

File: bootstrap/commandline/commandline.go [hunk 2]
```diff
@@ -36,8 +36,8 @@ type CommonFlags interface {
 	Help()
 }
 
-// DefaultCommonFlags is the Default implementation of CommonFlags used by most EdgeX services
-type DefaultCommonFlags struct {
+// Default is the Default implementation of Flags used by most EdgeX services
```
Comment (lenny-intel) [inline bootstrap/commandline/commandline.go]: Same issue here. `Default` what? `DefaultCommonFlags` tells the what.

File: bootstrap/commandline/commandline.go [hunk 3]
```diff
@@ -47,17 +47,21 @@ type DefaultCommonFlags struct {
 	configFileName    string
 }
 
-// NewDefaultCommonFlags creates and initializes a DefaultCommonFlags
-func NewDefaultCommonFlags(additionalUsage string) *DefaultCommonFlags {
-	commonFlags := DefaultCommonFlags{}
+// NewWithUsage returns a Default struct.
+func NewWithUsage(additionalUsage string) *Default {
```
Comment (lenny-intel) [inline bootstrap/commandline/commandline.go]: Again new what with usage?

File: bootstrap/commandline/commandline.go [hunk 4]
```diff
@@ -47,17 +47,21 @@ type DefaultCommonFlags struct {
...
+		FlagSet:         flag.NewFlagSet("", flag.ExitOnError),
+		additionalUsage: additionalUsage,
+	}
+}
 
-	commonFlags.FlagSet = flag.NewFlagSet("", flag.ExitOnError)
-	commonFlags.additionalUsage = additionalUsage
-	return &commonFlags
+// New returns a Default struct with an empty additional usage string.
+func New() *Default {
```
Comment (lenny-intel) [inline bootstrap/commandline/commandline.go]: Again new what?

File: bootstrap/commandline/commandline.go (hunk 1 above)
Comment (michaelestrin) [reply bootstrap/commandline/commandline.go]: > Also, per @tsconn23, bootstrap/commandline/commandline.go is name stuttering. I googled this and didn't find anything on filename stuttering. I did find package/struct stuttering guidelines, which isn't an issue here. ;-)

I have previously discussed this with @tsconn23 and convinced him that filename stutter isn't stutter we're concerned about.  See https://github.com/edgexfoundry/go-mod-bootstrap/blob/master/bootstrap/bootstrap.go for a representative existing example.

**Comment by michaelestrin** [comment]
Attempted to address @lenny-intel's concerns by strategically renaming commandline to flags.

**Pull request merged**